tasks/create_nat_gateway.yml

```diff
@@ -58,9 +65,49 @@
     - natgw.eip_name is defined
 
 
+# Idempotence Issue
+# -----------------
+#
+# ec2_vpc_nat_gateway returns a hard-coded ID in chek mode (nat-123456789)
+# this results in showing the playbook as 'changed' when doing a dry run.
+# To overcome this issue we do the following:
+#
+# 1. Gather existing NAT GW in that subnet
+# 2. If found:
+#    2.1. Use its ID for tagging
+# 3. If not found:
+#    3.1. Create it
+#    3.2. Gather facts again to find its id
```

Comment:
IIRC, its ID is returned when it's created, so point 3.2 would not be required. am I right?

Reply:
If not found and created in `check` mode, it still returns `nat-123456789` which will show the tagging task as changed.

Reply:
in that case, the module should not return bogus IDs when it detects that it's being ran in check mode. we should open an issue for this.